pkg/chunk/cache/fifo_cache.go

```diff
@@ -71,27 +71,30 @@ var (
 
 // FifoCacheConfig holds config for the FifoCache.
 type FifoCacheConfig struct {
-	Size     int           `yaml:"size"`
+	MaxSize  int           `yaml:"max_size_bytes"`
+	MaxCount int           `yaml:"size"`
 	Validity time.Duration `yaml:"validity"`
 }
 
 // RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
 func (cfg *FifoCacheConfig) RegisterFlagsWithPrefix(prefix, description string, f *flag.FlagSet) {
-	f.IntVar(&cfg.Size, prefix+"fifocache.size", 0, description+"The number of entries to cache.")
+	f.IntVar(&cfg.MaxSize, prefix+"fifocache.max-size-bytes", 0, description+"Maximum memory size of the cache.")
+	f.IntVar(&cfg.MaxCount, prefix+"fifocache.size", 0, description+"The number of entries to cache. (Deprecating, use 'fifocache.max-size-bytes' instead)")
 	f.DurationVar(&cfg.Validity, prefix+"fifocache.duration", 0, description+"The expiry duration for the cache.")
 }
 
 // FifoCache is a simple string -> interface{} cache which uses a fifo slide to
 // manage evictions.  O(1) inserts and updates, O(1) gets.
 type FifoCache struct {
 	lock     sync.RWMutex
-	size     int
+	maxCount int
+	maxSize  int
+	currSize int
 	validity time.Duration
-	entries  []cacheEntry
-	index    map[string]int
+	entries  map[string]*cacheEntry
```

Comment:
Could you explain why we need to change the data structure used for entries?

Reply:
explained below